build.zig

```diff
@@ -0,0 +1,175 @@
+        default_enable_threads = false;
+    }
+    const enable_threads = b.option(bool, "enable_threads", "Support threads") orelse default_enable_threads;
+    const enable_parallel_mark = b.option(bool, "enable_parallel_mark", "Parallelize marking and free list construction") orelse true;
+    const enable_thread_local_alloc = b.option(bool, "enable_thread_local_alloc", "Turn on thread-local allocation optimization") orelse true;
+    const enable_gcj_support = b.option(bool, "enable_gcj_support", "Support for gcj") orelse true;
+    const enable_java_finalization = b.option(bool, "enable_java_finalization", "Support for java finalization") orelse true;
+    const enable_atomic_uncollectable = b.option(bool, "enable_atomic_uncollectable", "Support for atomic uncollectible allocation") orelse true;
+    const enable_redirect_malloc = b.option(bool, "enable_redirect_malloc", "Redirect malloc and friend to GC routines") orelse false;
+    const enable_disclaim = b.option(bool, "enable_disclaim", "Support alternative finalization interface") orelse true;
+    const enable_large_config = b.option(bool, "enable_large_config", "Optimize for large heap or root set") orelse true;
+    const enable_mmap = b.option(bool, "enable_mmap", "Use mmap instead of sbrk to expand the heap") orelse true;
+    const enable_munmap = b.option(bool, "enable_munmap", "Return page to the OS if empty for N collections") orelse true;
+
+    const lib = b.addStaticLibrary(.{
+        .name = "gc",
+        .target = target,
+        .optimize = optimize,
+    });
+
+    var flags = std.ArrayList([]const u8).init(b.allocator);
+    defer flags.deinit();
+
+    if (t.abi.isMusl()) {
+        std.log.debug("Musl detected", .{});
+        flags.appendSlice(&.{"-DNO_GETCONTEXT"}) catch unreachable;
+    }
+
+    flags.appendSlice(&.{
+        "-fno-sanitize=undefined",
+        // Always enabled
+        "-DALL_INTERIOR_POINTERS",
+        "-DNO_EXECUTE_PERMISSION",
+        // Zig comes with clang that supports atomics so we can hardcode this
+        "-DGC_BUILTIN_ATOMIC",
+        // TODO: Should this be configurable?
+        "-DNO_PROC_FOR_LIBRARIES",
+    }) catch unreachable;
+
+    if (enable_threads) {
+        flags.appendSlice(&.{"-DGC_THREADS"}) catch unreachable;
+        if (enable_parallel_mark)
+            flags.appendSlice(&.{"-DPARALLEL_MARK"}) catch unreachable;
+        if (enable_thread_local_alloc) {
+            lib.addCSourceFiles(.{
+                .files = &.{
+                    "specific.c",
+                    "thread_local_alloc.c",
+                },
+                .flags = flags.items,
+            });
+        }
+        if (t.os.tag == .windows) {
+            lib.addCSourceFiles(.{
+                .files = &.{"win32_threads.c"},
+                .flags = flags.items,
+            });
+        } else {
+            lib.addCSourceFiles(.{
+                .files = &.{
+                    "gc_dlopen.c",
```

Comment:
Fixed!

Reply:
Zig is lacking some headers and stuff to make it work out of the box but I think it's possible to use if you supply your own cygwin threads library / headers etc. I saw an issue at some point about someone working on it, so it might be fixed. Anyway, it's aligned now!